components/identity/org.wso2.carbon.identity.certificateauthority/src/main/java/org/wso2/carbon/identity/certificateauthority/CAUserService.java

```
@@ -0,0 +1,118 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.certificateauthority;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.identity.certificateauthority.bean.CSR;
+import org.wso2.carbon.identity.certificateauthority.bean.Certificate;
+import org.wso2.carbon.identity.certificateauthority.internal.CAServiceComponent;
+import org.wso2.carbon.identity.certificateauthority.services.CSRService;
+import org.wso2.carbon.identity.certificateauthority.services.CertificateService;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+
+import java.util.List;
+
+/**
+ * This class contains the services that will be used by the CA users.
+ */
+@SuppressWarnings("UnusedDeclaration")
+public class CAUserService {
+
+    private static final Log log = LogFactory.getLog(CAUserService.class);
+    private CSRService csrService = CAServiceComponent.getCsrService();
+    private CertificateService certificateService = CAServiceComponent.getCertificateService();
+
+    /**
+     * Sends a CSR to the CA to be signed.
+     *
+     * @param encodedCSR PEM encoded CSR
+     * @return The serial number of the CSR that was stored at CA
+     * @throws CAException
+     */
+    public String addCSR(String encodedCSR) throws CAException {
+        String username = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        String userStoreDomain = UserCoreUtil.extractDomainFromName(username);
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        try {
+            return csrService.addCSR(encodedCSR, username, tenantDomain, userStoreDomain);
+        } catch (CAException e) {
+            log.error("Could not add CSR for user " + userStoreDomain + "\\" + username + "@" + tenantDomain, e);
+            throw new CAException("Could not store the CSR");
```

The original exception is swallowed here. Please fix.